File: tools/visualize_map.js
const mapData = require('../map/data/map_data.js'); // Wait, this is window.MAP_DATA in file.

// We need to parse the file since it's not a valid CommonJS module.
const fs = require('fs');
const content = fs.readFileSync('../map/data/map_data.js', 'utf8');
const jsonContent = content.replace('window.MAP_DATA = ', '');
const tiles = JSON.parse(jsonContent);

let maxX = 0;
let maxY = 0;
tiles.forEach(t => {
    if (t.gx > maxX) maxX = t.gx;
    if (t.gy > maxY) maxY = t.gy;
});

const grid = Array(maxY + 1).fill().map(() => Array(maxX + 1).fill('   '));

tiles.forEach(t => {
    // We can just print tx,ty
    // grid[t.gy][t.gx] = `${t.tx},${t.ty}`;
    grid[t.gy][t.gx] = `[${t.tx},${t.ty}]`;
});

// Print in chunks
console.log(`Map Size: ${maxX + 1} x ${maxY + 1}`);
for (let y = 0; y <= maxY; y++) {
    // console.log(grid[y].join(''));
    // Simplify: look for unique tiles.
    // Assume floor is 1,0 or 22,0.
    let row = '';
    for (let x = 0; x <= maxX; x++) {
        const t = tiles.find(tile => tile.gx === x && tile.gy === y);
        if (!t) { row += ' ... '; continue; }
        if (t.tx === 1 && t.ty === 0) row += ' . ';
        else if (t.tx === 22 && t.ty === 0) row += ' # ';
        else row += `[${t.tx},${t.ty}]`; // Mark interesting tiles
    }
    console.log(`${y}: ${row}`);
}

File: tools/visualize_map.py
import json
import sys

try:
    with open('map/data/map_data.js', 'r') as f:
        content = f.read()
        # manual parsing since it's "window.MAP_DATA = [...]"
        json_str = content.split('=', 1)[1].strip()
        # Remove trailing semicolon if exists
        if json_str.endswith(';'):
            json_str = json_str[:-1]
        
        tiles = json.loads(json_str)
        
        # Grid range to view: x=10-25, y=15-30
        for y in range(15, 30):
            row_str = f"{y:2d}: "
            for x in range(10, 26):
                # Find tile
                found = None
                for t in tiles:
                    if t['gx'] == x and t['gy'] == y:
                        found = t
                        break
                
                if found:
                    if found['tx'] == 1 and found['ty'] == 0:
                        row_str += " .  "
                    elif found['tx'] == 22 and found['ty'] == 0:
                        row_str += " #  "
                    else:
                        row_str += f"[{found['tx']},{found['ty']}]"
                else:
                    row_str += "    "
            print(row_str)

except Exception as e:
    print(e)
